### PR TITLE
Remove 2to3 in favour using six

### PR DESCRIPTION
**Pull request by lambacck**
Support Python 2.6-3.5 using Python 2/3 compatible codebase using six.

### CODE REVIEW

**Comment by lambacck** [comment]
Tests pass. Anyone feel like reviewing this? It's a pretty big change.

**Comment by Cito** [comment]
I think it's a good idea in the first place. 2to3 was the recommended way a few years ago, but it has several drawbacks. Will only have time to review over the winter holidays. Maybe you can release it as beta first if you are not sure.

**Comment by Cito** [comment]
Looks good to me.

Maybe instead of  `import six.moves.html_parser` and then using it as `six.moves.html_parser.HTMLParser(...)` we should `from six.moves.html_parser import HTMLParser` and then using it as `HTMLParser(...)`? This would remove the `six.` clutter from the actual code, which makes it shorter and easier to read and also easier to clean up when we will drop Python 2 support one day.

Also, we should add the exact supported Py versions as Trove classifiers in setup.py.

**Comment by lambacck** [comment]
I updated the imports and the trove classifiers. We're not currently testing 3.2 but I think it might work. We could simplify a bunch of things by adding u prefix back to unicode strings if we decided that 3.2 wasn't worth it. https://caremad.io/2015/04/a-year-of-pypi-downloads/ seems to indicate that the use of python 3.2 is vanishingly small but that would have us matching Pyramid's version support for instance.

**Comment by Cito** [comment]
You can use `six.u` instead of the `u` prefix, but it can be problematic with non-ascii strings which may be used in the tests. And I wouldn't mind dropping support for Py 3.2 - Requests, Jinja2, Ming, Coverage don't support it either.

**Comment by lambacck** [comment]
I've removed six.u in favour of u''. If all the tests pass then I'll call this done and merge it. Then I'll look into trying to merge other pull requests in an effort to get us to a new release.

**Comment by nZac** [comment]
Is there anything holding this back/something that I can do to help it get released? #104 is causing me some issues. Thanks!

**Comment by lambacck** [comment]
Just my time to do a release. I think I had ambitions to merge some more things.

File: formencode/tests/test_validators.py [hunk 1]
```diff
@@ -10,6 +12,10 @@
 from formencode.schema import Schema
 from formencode.foreach import ForEach
 from formencode.api import NoDefault
+import six
+
+if not six.PY2:
```
Comment (nZac) [inline formencode/tests/test_validators.py]: I am not seeing `unicode` used anywhere in this file.

File: formencode/tests/test_compound.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
```
Comment (3noch) [inline formencode/tests/test_compound.py]: This appears before the UTF encoding marker. Is that ok?

File: formencode/tests/test_cc.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
```
Comment (3noch) [inline formencode/tests/test_cc.py]: This does too.

**Comment by nZac** [comment]
Hi @lambacck sorry to bug you on this again. Any chance of this getting merged and released soon, it is one of the only dependencies holding me back from going Py3.

**Comment by lambacck** [comment]
@nZac the current released version formencode already supports python 3.

**Comment by nZac** [comment]
@lambacck, thanks for merging this. #104 is what is tripping me up. Though, I will try switching to the alpa version and see if that helps.

**Pull request merged**